File: python/OS_algorithms/fcfs.py
# Python program to demonstrate
# FCFS Disk Scheduling algorithm

size = 8


def FCFS(arr, head):

    seek_count = 0
    distance, cur_track = 0, 0

    for i in range(size):
        cur_track = arr[i]

        # calculate absolute distance
        distance = abs(cur_track - head)

        # increase the total count
        seek_count += distance

        # accessed track is now new head
        head = cur_track

    print("Total number of seek operations = ", seek_count)

    # Seek sequence would be the same
    # as request array sequence
    print("Seek Sequence is")

    for i in range(size):
        print(arr[i])


# Driver code
if __name__ == "__main__":

    # request array
    arr = [176, 79, 34, 60, 92, 11, 41, 114]
    head = 50

    print("Initial position of head:", head)
    
    FCFS(arr, head)

# This code contributed by Rajput-Ji

File: bombies/OS_algorithms/fcfs.cpp
// FCFS Disk Scheduling algorithm

#include <bits/stdc++.h>
using namespace std;

void FCFS(int arr[], int size, int head);

//****************************************************************************************************

int main() {
    int size,
        head;
    // request array

    cout << "FCFS" << endl;
    cout << "How many request: ";
    cin >> size;
    cout << "Read/Write head starts on track: ";
    cin >> head;

    int* arr = new int[size];
    for (int i = 0; i < size; ++i) {
        cout << "Enter request " << i + 1 << ": ";
        cin >> arr[i];
    }

    FCFS(arr, size, head);

    delete[] arr;
    return 0;
}

//****************************************************************************************************

void FCFS(int arr[], int size, int head) {
    int seek_count = 0;
    int distance, cur_track;

    for (int i = 0; i < size; i++) {
        cur_track = arr[i];

        // calculate absolute distance
        distance = abs(cur_track - head);

        // increase the total count
        seek_count += distance;

        // accessed track is now new head
        head = cur_track;
    }

    cout << "\nTotal number of seek operations = "
         << seek_count << endl;
    cout << "Average seek: " << (static_cast<double>(seek_count) / size) << endl;
    // Seek sequence would be the same
    // as request array sequence
    cout << "Seek Sequence is" << endl;

    for (int i = 0; i < size; i++) {
        cout << arr[i] << endl;
    }
}